bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Control.java

```diff
@@ -711,6 +713,9 @@ void createWidget () {
 	if ((state & PARENT_BACKGROUND) != 0) {
 		setBackground ();
 	}
+	if((style & SWT.EMBEDDED) != 0) {
```

Comment:
> Do all 'embedded' controls actually have this problem?

AFAIK, yes. You can try embedding an Excel sheet, it'll flicker too.

> Is this workaround actually needed still? I removed it for testing and cannot see any problem.

Since the logic checks for the presence of `SWT.EMBEDDED`, if you remove it then Edge will flicker. I tried and it flickers (I had to restart the application after the change, hot-code replacement didn't work).

Reply:
My understanding of the original question (whether the _original_ workaround is still needed) asks for the flag being required as a workaround for the focus handling, not for identifying that for the control redrawing should not be disabled.\
So rephrasing the question: with this PR, does setting the embedded flag become a workaround just for identifying that redrawing should not be disabled for the browser control? If that's the case, we might think of other properties of all controls for which redrawing should not be disabled.

Reply:
I don't know if `SWT.EMBEDDED` is still needed for the browser to receive focus (according to @sratz, it isn't) but it is needed for this PR.

What other identifier would you propose instead?

Reply:
> but it is needed for this PR.

That's not necessarily true. From a conceptual perspective, what is needed for this PR is some identifier for controls for which redrawing must not be disabled because they simply disappear when disabling redrawing (and then lead to flickering). The current assumption was that `SWT.EMBEDDED` is a valid identifier, but if Edge just uses this flag as some workaround (that we may remove now), it does not seem to be a proper identifier but is rather another workaround to attach some (rather random) flag to controls for identifying that redrawing should not be disabled for them. If I remember correctly, this embedded flag is also used for embedding Swing controls, isn't it? And is it correct to disable redrawing for them as well?

Might be that there currently is no proper identifier for those controls for which redrawing must not be disabled. Then a solution might be to add some method to `Control` to identify whether redrawing can be disabled, which can then be properly implemented for the affected controls like a browser using Edge (and other embeddings of native controls on Windows).

Reply:
I get what you guys mean now, thank you for the clarification. 

I changed the code so it doesn't use `SWT.EMBEDDED` but a newly created method instead.